test/gallery-route-test.js

```diff
@@ -0,0 +1,238 @@
+    });
+    describe('Invalid Request', () => {
+      it('should return 404', done => {
+        request.get(`${url}/api/gallery/12345`)
+          .set({ Authorization: `Bearer ${this.tempToken}` })
+          .end( res => {
+            expect(res.status).to.equal(404);
+            done();
+          });
+      });
+    });
+    describe('Missing Token', () => {
+      it('should return 404', done => {
+        request.get(`${url}/api/gallery/${this.tempGallery._id}`)
+          .end( res => {
+            expect(res.status).to.equal(401);
+            done();
+          });
+      });
+    });
+  });
+
+  describe('GET: /api/gallery', () => {
+    before( done => {
+      testGallery.userID = this.tempUser._id.toString();
+      new Gallery(testGallery).save()
+        .then( gallery => {
+          this.tempGallery = gallery;
+          done();
+        })
+        .catch(done);
+    });
+    after( () => {
+      delete testGallery.userID;
+    });
+    describe('Valid Body', () => {
+      it('should return all galleries', done => {
+        request.get(`${url}/api/gallery`)
+          .set({ Authorization: `Bearer ${this.tempToken}` })
+          .end( (err, res) => {
+            if(err) return done(err);
+            expect(res.status).to.equal(200);
+            done();
+          });
+      });
+    });
+  });
+
+  describe('PUT: /api/gallery/:id', () => {
+    before( done => {
+      testGallery.userID = this.tempUser._id.toString();
+      new Gallery(testGallery).save()
+        .then( gallery => {
+          this.tempGallery = gallery;
+          done();
+        })
+        .catch(done);
+    });
+    after( () => {
+      delete testGallery.userID;
+    });
+    describe('Valid Body', () => {
+      it('should return a new gallery', done => {
+        var updated = { name: 'Updated Name', description:'updated description' };
+        request.put(`${url}/api/gallery/${this.tempGallery._id}`)
+          .send(updated)
+          .set({ Authorization: `Bearer ${this.tempToken}` })
+          .end( (err, res) => {
+            if(err) return done(err);
+            expect(res.status).to.equal(200);
+            expect(res.body.name).to.equal(updated.name);
+            done();
+          });
+      });
+    });
+    describe('Missing Token', () => {
+      it('should return a new gallery', done => {
+        var updated = { name: 'Updated Name' };
+        request.put(`${url}/api/gallery/${this.tempGallery._id}`)
+          .send(updated)
+          .end( res => {
+            expect(res.status).to.equal(401);
+            done();
+          });
+      });
+    });
+    describe('Invalid Body', () => {
+      it('should return a new gallery', done => {
+        var brokenUpdate = { name: 5 };
+        request.put(`${url}/api/gallery/${this.tempGallery._id}`)
+          .send(brokenUpdate)
+          .set({ Authorization: `Bearer ${this.tempToken}` })
+          .end( res => {
+            expect(res.status).to.equal(400);
```

Comment:
Make sure the it block contains the right string to correspond with what you are testing for 👍  In this case, it is misleading since you're actually going to be looking for a bad request.